### ISSUE TITLE
Debian testing: libssl.so.1.0.0 not found

### ISSUE THREAD

**Issue by cblichmann** [Vector35/binaryninja-api#878]
Just dowloaded and extracted the latest Linux version:
```
$ ./binaryninja                                                                      
./binaryninja: error while loading shared libraries: libssl.so.1.0.0: cannot open shared object file: No such file or directory
```

This version links against `libssl.so.1.0.2`. However, my (really old) version 1.0.320 from March this year works and links against `libssl.so.1.0.2`.

OS is Debian Testing

**Comment by GeneralBison** [comment]
I am having this same issue on Debian Stretch. Even when following the [troubleshooting](https://docs.binary.ninja/guide/troubleshooting/index.html#debian).

I used strace to attempt to figure out why BinaryNinja isn't seeing the lib and it doesn't look like it's looking in the plugins directory:

```
open("/home/mb/bin/binaryninja/libssl.so.1.0.0", O_RDONLY|O_CLOEXEC) = -1 ENOENT (No such file or directory)
open("/home/mb/bin/binaryninja/libssl.so.1.0.0", O_RDONLY|O_CLOEXEC) = -1 ENOENT (No such file or directory)
access("/etc/ld.so.nohwcap", F_OK)      = -1 ENOENT (No such file or directory)
open("/lib/x86_64-linux-gnu/tls/x86_64/libssl.so.1.0.0", O_RDONLY|O_CLOEXEC) = -1 ENOENT (No such file or directory)
stat("/lib/x86_64-linux-gnu/tls/x86_64", 0x7fff72125f10) = -1 ENOENT (No such file or directory)
open("/lib/x86_64-linux-gnu/tls/libssl.so.1.0.0", O_RDONLY|O_CLOEXEC) = -1 ENOENT (No such file or directory)
stat("/lib/x86_64-linux-gnu/tls", 0x7fff72125f10) = -1 ENOENT (No such file or directory)
open("/lib/x86_64-linux-gnu/x86_64/libssl.so.1.0.0", O_RDONLY|O_CLOEXEC) = -1 ENOENT (No such file or directory)
stat("/lib/x86_64-linux-gnu/x86_64", 0x7fff72125f10) = -1 ENOENT (No such file or directory)
open("/lib/x86_64-linux-gnu/libssl.so.1.0.0", O_RDONLY|O_CLOEXEC) = -1 ENOENT (No such file or directory)
stat("/lib/x86_64-linux-gnu", {st_mode=S_IFDIR|0755, st_size=12288, ...}) = 0
```

I've attached the full strace output.
[bnstrace.txt](https://github.com/Vector35/binaryninja-api/files/1491855/bnstrace.txt)

**Comment by plafosse** [comment]
I'm currently reworking how we link against libssl, libcrypto, and libssl. Hopefully, I'll soon have a solution that will cause a lot less headaches for people that aren't running on Ubuntu.

**Comment by GeneralBison** [comment]
Don't know what causes this problem but a workaround is sym-linking the libraries in the binaryninja directory.
e.g.
`ln -s binaryninja/plugins/libssl.so binaryninja/libssl.so.1.0.0`

**Comment by psifertex** [comment]
Yup! That work around is documented here: 

https://docs.binary.ninja/guide/troubleshooting/index.html#linux

**Comment by GeneralBison** [comment]
@psifertex  Not quite, the workaround specifies that the links be placed in the plugins directory, I found that the links have to be in the parent directory (binaryninja).

My strace shows that it's not checking the plugins directory.

**Comment by psifertex** [comment]
Ahh, gotcha. Thanks, I'll update it.

**Comment by psifertex** [comment]
Though as Peter mentioned -- he's working on the proper long-term fix.

**Comment by plafosse** [comment]
Please check out the dev channel, which should have a fix for this issue as of build 1.1.1009. Please reopen if this issue doesn't seem to be fixed.
